### PR TITLE
Make scanner resilient to binary files as input.

### LINKED ISSUE #2867: VB parser runs out of memory when given a VSIX file as text
It was reported at https://github.com/dotnet/roslyn/issues/2771#issuecomment-102195182 that the VB parser exhausts memory when passed a VSIX file as input. We should consider the same fix for VB as any fix that we apply for C#.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #2867.

@VSadov, @gafter, @agocke, @jaredpar, @khyperia Please review.

### CODE REVIEW

**Comment by VSadov** [comment]
LGTM

File: src/Compilers/VisualBasic/Test/Syntax/Parser/FuzzTesting.vb [hunk 1]
```diff
@@ -918,4 +918,22 @@ End Enum
...
+        Dim builder = New System.Text.StringBuilder(n + 10)
+        For i As Integer = 0 To n - 1
+            builder.Append(ChrW(random.Next(&HFFFF)))
+        Next
+
+        Dim source = builder.ToString()
+        Dim tree = VisualBasicSyntaxTree.ParseText(source)
+
+        Assert.Equal(source, tree.ToString())
+        Assert.Equal(200, tree.GetDiagnostics().Where(Function(d) d.Code = ERRID.ERR_IllegalChar).Count())
```
Comment (jaredpar) [inline src/Compilers/VisualBasic/Test/Syntax/Parser/FuzzTesting.vb]: Can we make 200 a constant given that it's used in 3 different places now?

**Comment by jaredpar** [comment]
:+1:

File: src/Compilers/VisualBasic/Test/Syntax/Parser/FuzzTesting.vb (hunk 1 above)
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Test/Syntax/Parser/FuzzTesting.vb]: Sure, I will add a constant.

**Comment by AlekseyTs** [comment]
- Bug number - Fixes #2867
- Customer scenario - An attempt to parse a big file, which doesn't contain VB code (a binary file, for example), can cause a huge number of errors produced by the scanner. This in turn can consume huge amount of memory, eventually causing an out of memory exception.  
- Fix description - Limit the number of invalid tokens encountered to 200. Once the limit is reached, the rest of the file is skipped by the scanner, the remaining content is preserved in the last “bad” token.
- Testing done - unit-tests.

@jaredpar FYI

**Comment by MattGertz** [comment]
Approved 5/27 by ML Shiproom.

**Comment by gafter** [comment]
LGTM

**Pull request merged**